packages/core/src/lib/compare.unit.test.ts

```diff
@@ -0,0 +1,137 @@
+    });
+
+    it('should have no changes, additions or removals', () => {
+      const reportsDiff = compareReports(mockReports);
+      expect(reportsDiff.categories.changed).toHaveLength(0);
+      expect(reportsDiff.categories.added).toHaveLength(0);
+      expect(reportsDiff.categories.removed).toHaveLength(0);
+      expect(reportsDiff.groups.changed).toHaveLength(0);
+      expect(reportsDiff.groups.added).toHaveLength(0);
+      expect(reportsDiff.groups.removed).toHaveLength(0);
+      expect(reportsDiff.audits.changed).toHaveLength(0);
+      expect(reportsDiff.audits.added).toHaveLength(0);
+      expect(reportsDiff.audits.removed).toHaveLength(0);
+    });
+
+    it('should contain all categories/groups/audits in unchanged arrays', () => {
+      const reportsDiff = compareReports(mockReports);
+      expect(reportsDiff.categories.unchanged).toHaveLength(
+        mockReport.categories.length,
+      );
+      expect(reportsDiff.groups.unchanged).toHaveLength(
+        mockReport.plugins.reduce((acc, { groups }) => acc + groups!.length, 0),
+      );
+      expect(reportsDiff.audits.unchanged).toHaveLength(
+        mockReport.plugins.reduce((acc, { audits }) => acc + audits.length, 0),
+      );
+    });
+  });
+
+  describe('changed reports', () => {
+    const mockReports: Diff<Report> = {
+      before: { ...MINIMAL_REPORT_MOCK, commit: mockCommits.before },
+      after: { ...REPORT_MOCK, commit: mockCommits.after },
+    };
+
+    it('should create valid report diff', () => {
+      const reportsDiff = compareReports(mockReports);
+      expect(() => reportsDiffSchema.parse(reportsDiff)).not.toThrow();
+    });
+
+    it('should include commits from both reports', () => {
+      const reportsDiff = compareReports(mockReports);
+      expect(reportsDiff.commits).toEqual(mockCommits);
+    });
+
+    it('should only have added categories (minimal report has none)', () => {
+      const reportsDiff = compareReports(mockReports);
+      expect(reportsDiff.categories.added).toHaveLength(
+        REPORT_MOCK.categories.length,
+      );
+      expect(reportsDiff.categories.removed).toHaveLength(0);
+      expect(reportsDiff.categories.changed).toHaveLength(0);
+      expect(reportsDiff.categories.unchanged).toHaveLength(0);
+    });
+
+    it('should only have added groups (minimal report has none)', () => {
+      const reportsDiff = compareReports(mockReports);
+      expect(reportsDiff.groups.added.length).toBeGreaterThan(0);
+      expect(reportsDiff.groups.removed).toHaveLength(0);
+      expect(reportsDiff.groups.changed).toHaveLength(0);
+      expect(reportsDiff.groups.unchanged).toHaveLength(0);
+    });
+
+    it('should mark multiple audits as added and 1 as removed (single audit from minimal report unmatched)', () => {
+      const reportsDiff = compareReports(mockReports);
+      expect(reportsDiff.audits.added.length).toBeGreaterThan(1);
+      expect(reportsDiff.audits.removed).toHaveLength(1);
```

Comment:
I would at least check its `slug`. Again, static mock, so we know which one it is.

Reply:
Right. So if you are only concerned if there are any such entities present, why single out this instance where you expect one - then the question becomes, how do you know if it is the right one? If you only care that there is at least one in order not to disrupt the test data, I still think `.tobeGreaterThan(0)` is the correct approach here. `toHaveLength(N)` is neither here or there.

Reply:
I missed the commit where you addressed this so feel free to ignore, this is outdated, sorry 😅